docs/adr/adr_0012.md

```diff
@@ -0,0 +1,129 @@
+
+### `ClusterParseDefinition`
+
+Other then being cluster scoped `ClusterParseDefinition`s are identical to the existing `ParseDefinition` CRD.
+
+When a scan job has completed and the results need to be parsed, the operator should first look for `ParseDefinition`s in the `Scan`s namespace matching the `scantype.spec.extractResults.type` / `clusterscantype.spec.extractResults.type` name configured in the ScanType.
+If that is not found the operator should look for `ClusterParseDefinition`s with the same name.
+No matter which was picked the `Job` for the parser always has to be started in the namespace of the `Scan`.
+
+### `ClusterCascadingRule`
+
+Other then being cluster scoped `ClusterCascadingRule`s are identical to the existing `CascadingRule` CRD.
+
+When the cascading scans hook starts, the hook should fetch the list of both `CascadingRule`s and `ClusterCascadingRule`s and merge them together into one list which then get's evaluated against the scan just like it is now. All scans started from it have to be in the same namespace of the original `Scan`.
+
+#### Deduplication of CascadingRules
+
+One potential issue with CascadingRule and ClusterCascadingRule is that in contrast to ScanType and ParseDefinition both namespaced and cluster wide types are used. This can be problematic when both the namespace as well as the cluster level have similar / equal CascadingRule installed as this could lead to both producing the same scan which would then be executed twice.
+
+Potential workarounds here would be to introduce an additional field to the CascadingRule spec which allows us to define a sort of "deduplication" / replacement mechanism.
+
+### `ClusterScanCompletionHook`
+
+`ClusterScanCompletionHook` behave mostly like their namespaced counterpart, when the parser completes the operator fetches the list of ScanCompletionHooks from Scans namespace (respecting the hookSelector of the Scan) and the `ClusterScanCompletionHook`s (also respecting the hookSelector of the Scan (this might have to be discussed...)) are merged and ordered according to the respective priorities set on them.
+
+The execution of the hooks is different to the current model, as there be an additional setting on the `ClusterScanCompletionHook` called namespaceMode which allows users to configure to either run the hook in the same namespace as the scan it is getting executed for, or to provide a fixed namespace where all executions for this `ClusterScanCompletionHook` will be run in.
+
+Example `ClusterScanCompletionHook` specs with this being set:
+
+```yaml
+apiVersion: "execution.securecodebox.io/v1"
+kind: ClusterScanCompletionHook
+metadata:
+  name: slack-notification
+spec:
+  type: ReadAndWrite
+  image: "docker.io/securecodebox/notification"
+```
```

Comment:
Is this yaml example necessary? It does not specify the execution namespace. If it is meant to be an example for the same namespace as the scan, I would add a comment to clarify that.